js/windowEventRecorder.js

```diff
@@ -0,0 +1,72 @@
+const windowEventsToRecord = {
+    CLICK: 'click',
+    DBLCLICK: 'dblclick',
+    CHANGE: 'change',
+    SELECT: 'select',
+    SUBMIT: 'submit'
+}
+
+let enabled = false;
+
+const sendMessage = (msg) => {
+    try {
+        browser.runtime.sendMessage({
+            type: 'loadster_browser_event',
+            value: msg
+        })
+    } catch (err) {
+        console.debug('caught error', err)
+    }
+}
+
+const recordEvent = (e) => {
+    // we explicitly catch any errors and swallow them, as none node-type events are also ingested.
+    // for these events we cannot generate selectors, which is OK
+    try {
+        const optimizedMinLength = (e.target.id) ? 2 : 10 // if the target has an id, use that instead of multiple other selectors
+        const selector = finder(e.target, { seedMinLength: 5, optimizedMinLength: optimizedMinLength })
+        const msg = {
+            selector: selector,
```

Comment:
A list of tag attributes might be nice here too (like `id`, `class`, `name`, etc).